nodes/PayloadValidator.js

```diff
@@ -41,40 +42,89 @@ module.exports = class PayloadValidator {
     }
   }
 
+  /**
+   * Gets the value at the given location in the given object
+   *
+   * @param {*} object
+   * @param {*} location
+   * @returns
+   */
   getValue(object, location) {
-    return location.split('.').reduce((p, c) => (p && p[c]) || null, object);
+    return _.get(object, location);
+  }
+
+  /**
+   * Sets the value at the given location in the given object to the given value
+   *
+   * @param {*} object
+   * @param {*} location
+   * @param {*} value
+   * @returns
+   */
+  setValue(object, location, value) {
+    return _.set(object, location, value);
+  }
+
+  /**
+   * lodash set will always pass but might not actually set the value
+   * So here we test that it was actually set and use that to confirm we can set
+   *
+   * @param {*} object
+   * @param {*} location
+   * @returns
+   */
+  canSet(object, location) {
```

Comment:
can you use lodash `has` instead of this?

Reply:
Not really, since we need to make sure we can set on the location, since at any point of the path could of be modified to be a string for example


``` js
const after = {
  payload: {
    system: 'some string'
  }
};

// And we are trying to set
// after.payload.system.organization;

// they could also do something like

const after = {} // this will allow us to set org back correctly
```